### DISCUSSION TITLE
get_total_irradiance vs single_axis_tracker.get_irradiance

### DISCUSSION THREAD

**Discussion by cpullium** [pvlib/pvlib-python#1555] [Q&A] [answered]
Hey all,
Just started using pvlib for the first time the last couple of months. Very cool!

I'm trying to generate some time series POA. For fixed tilt, I've been using `get_total_irradiance()`. That works well. 
However, when I tried to use `tracking.singleaxis()` to get tracker angles, and input the tracker_theta return into `get_total_irradiance()`, I'm getting some funny looking results. 

Is `tracking.singleaxis()` the recommended method? Should I be using `SingleAxisTracker.get_irradiance()`?? What's the difference?
I could share code, and get in the weeds, but I'm just looking for a general understanding here. 

Thanks everyone!

**Comment by kandersolar** [comment, marked as answer]
Howdy @cpullium, it's always important make sure that the values you're passing in match the required parameters.  For example, the docs page for [get_total_irradiance](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.get_total_irradiance.html) lists `surface_tilt` and `surface_azimuth` (not `tracker_theta`) as parameters.  Conceptually, `get_total_irradiance` is agnostic to any mounting strategy -- what matters for transposition is the collector surface's tilt & azimuth, and it doesn't make any difference whether that orientation came from a fixed rack or a SAT.  Note that the dataframe returned from `tracking.singleaxis()` contains `surface_tilt` and `surface_azimuth` in addition to `tracker_theta`. 

As for whether to use `tracking.singleaxis` or `SingleAxisTracker` -- pvlib's foundation is a set of stand-alone functions like `tracking.singleaxis` that you can string together to do whatever you want, and built atop that foundation is a collection of classes that might be more convenient for typical use cases.  I usually stick to just using the functions myself, but either way is fine and I suggest you use whichever approach you find easier to use.  The [intro tutorial](https://pvlib-python.readthedocs.io/en/stable/user_guide/introtutorial.html) page goes into more detail about the two approaches.  However, in the specific case of `SingleAxisTracker`, that class is actually [deprecated](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.tracking.SingleAxisTracker.html) and should not be used in new code.  If you do want to go the OO route, the new way is to use a `SingleAxisTrackerMount` with `PVSystem` like is done in the [PVSystem](https://pvlib-python.readthedocs.io/en/stable/user_guide/pvsystem.html) user guide page.

**Reply by cpullium** [reply]
Thanks @kanderso-nrel, I was indeed using the wrong input.
